fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/DefaultOCFLObjectSession.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.persistence.ocfl.impl;
+
+import static org.fcrepo.persistence.ocfl.api.CommitOption.NEW_VERSION;
+import static org.fcrepo.persistence.ocfl.api.CommitOption.MUTABLE_HEAD;
+
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.fcrepo.persistence.ocfl.api.CommitOption;
+import org.fcrepo.persistence.ocfl.api.OCFLObjectSession;
+
+import edu.wisc.library.ocfl.api.OcflRepository;
+import edu.wisc.library.ocfl.api.model.ObjectId;
+
+/**
+ * Default implementation of an OCFL object session, which stages changes to the
+ * file system prior to committing.
+ *
+ * @author bbpennel
+ */
+public class DefaultOCFLObjectSession implements OCFLObjectSession {
+
+    private String objectIdentifier;
+
+    // Path where changes to the OCFL object in this session are staged
+    private Path stagingPath;
+
+    private Set<String> deletePaths;
+
+    private OcflRepository ocflRepository;
```

Comment:
This should be `MutableOcflRepository` if you want the Mutable HEAD related APIs.